File: 4 занятие/3B_home.py
import collections

a = [collections.deque() for i in range(150001)]

n = int(input())
for i in range(n):
    t = input().split()

    if t[0] == "pushback":
        a[int(t[1])].append(int(t[2]))
    elif t[0] == "pushfront":
        a[int(t[1])].appendleft(int(t[2]))
    elif t[0] == "popback":
        print(a[int(t[1])].pop())
    elif t[0] == "popfront":
        print(a[int(t[1])].popleft())

File: 4 занятие/3B_home.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    deque<int> a[150001];

    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        string s;
        cin >> s;
        int a1, a2;
        if (s == "pushback") {
            cin >> a1 >> a2;
            a[a1].push_back(a2);
        } else if (s == "pushfront") {
            cin >> a1 >> a2;
            a[a1].push_front(a2);
        } else if (s == "popback") {
            cin >> a1;
            cout << a[a1].back() << "\n";
            a[a1].pop_back();
        } else {
            cin >> a1;
            cout << a[a1].front() << "\n";
            a[a1].pop_front();
        }
    }
}
